vertical-pod-autoscaler/pkg/recommender/input/history/history_provider_test.go

```diff
@@ -257,6 +262,25 @@ func TestGetMemorySamples(t *testing.T) {
 	assert.Equal(t, histories, map[model.PodID]*PodHistory{podID: podHistory})
 }
 
+func TestAuthHeader(t *testing.T) {
+	historyProvider, err := NewPrometheusHistoryProvider(getDefaultPrometheusHistoryProviderConfigForTest())
+	if err != nil {
+		t.Fatal(err)
+	}
+	authHeaderName := "Token"
+	testToken := "aaaaaaaaaaaa"
+	tokenEnvVariableName := "TEST_TOKEN"
+	os.Setenv(tokenEnvVariableName, testToken)
+
+	req, _ := http.NewRequest("GET", "http://example.com", nil)
+
+	_, err = historyProvider.GetClusterHistory()
+
+	assert.Nil(t, err)
+	assert.Equal(t, testToken, req.Header.Get(authHeaderName))
+	defer os.Unsetenv(tokenEnvVariableName)
```

Comment:
I don't think this correctly unsets the env if the asserts above fail. Maybe consider using https://pkg.go.dev/testing#T.Setenv instead or moving the defer to where the env is set.

Reply:
done @kwiesmueller